puppet/modules/jenkins_job_builder/files/centos.org/pipelines/test/foremanNightlyTest.groovy

```diff
@@ -0,0 +1,42 @@
+pipeline {
+    agent { label 'foreman' }
+
+    stages {
+        stage('Setup Environment') {
+            steps {
+                deleteDir()
+                git url: 'https://github.com/theforeman/forklift.git'
+            }
+        }
+        stage('Provision Node') {
+            steps {
+                provision()
+            }
+        }
+        stage('Install Pipeline Requirements') {
+            steps {
+                runPlaybook('playbooks/setup_forklift.yml', cico_inventory('./'), [], ['-b'])
+            }
+        }
+        stage('Run Pipeline') {
+            steps {
+		duffy_ssh("cd forklift && ansible-playbook pipelines/bats_pipeline_foreman_nightly.yml -e 'forklift_state=up'", 'duffy_box', './')
+            }
+        }
+    }
+
+    post {
+        always {
+            script {
+                duffy_ssh('cd forklift && ansible-playbook playbooks/collect_debug.yml -l centos7-foreman-bats-ci', 'duffy_box', './')
+                runPlaybook('foreman-infra/ci/centos.org/ansible/fetch_debug_files.yml', cico_inventory('./'), ["workspace=/home/foreman/workspace/${env.JOB_NAME}/debug"], ['-b'])
```

Comment:
Maybe I'm missing something but how can this job find `foreman-infra`? Should it clone that git repository?

Reply:
It's a fair question. foreman-infra is downloaded in the `provision` function. That is an unnecessary obfuscation, I'll go ahead and make a separate PR to  unbind that and make it more clear.

Reply:
I just decided to do this within this PR.

Reply:
Did you revert it again?